### ISSUE TITLE
Restructuring for Adding New Notification Methods as Plugins

### ISSUE THREAD

**Issue by The-EpaG** [SpectralVectors/RenderReminder#4]
I would like to propose a modification to the structure of the SpectralVectors/RenderReminder add-on to simplify the addition of new notification methods, such as Discord, web-hook, etc., by transforming them into plugins. I have already created a proof of concept (POC) to demonstrate how this structural change could work.

**Current Issue:**
Currently, adding new notification methods would require substantial modifications to the main source code of the add-on. This makes the extension process less flexible and potentially prone to errors.

**Proposed Modification:**
My proposal is to restructure the add-on so that notification methods are implemented as separate plugins. This would allow developers to add new notification methods without having to modify the main source code of the add-on, making the process more modular and scalable.

**Proof of Concept:**
I have already created a [POC](https://github.com/The-EpaG/RenderReminder-poc) that demonstrates how this new structure could work in terms of implementation. The POC includes examples of how plugins for Discord and web-hook could be integrated into the add-on independently.

**Request:**
I ask the project maintainers if they agree with this proposed modification and if they are interested in reviewing the POC I have created. I am available for further discussion and collaboration to implement this modification.

Thank you for your attention.

**Comment by SpectralVectors** [comment]
This is a very interesting idea, and one that would help support the broadest range of services. 

I realize that my code in this case is rather poorly organized, and certainly wasn't initially designed with plug-in system in mind.

I'm open to implementing this, but looking at the POC I didn't see a `bl_info` dict, registered classes or some of the other boilerplate code that goes into Blender addons. 

Is the POC just to demonstrate how the plug-in system is set up, or can this actually install and run in Blender?

Very interesting, thank you for your thoughts and contributions!

**Comment by The-EpaG** [comment]
Thank you for considering my proposal and for the positive feedback!

The POC I've created is designed to demonstrate the functioning of the plugin system, rather than to be directly installed into Blender. I can confirm that it currently does not include the necessary details for installation and direct execution within the Blender environment, such as the _bl_info_ dict.

However, I can provide further clarification on the structure and functioning of the POC through a code review. I'm available to explain in detail how I designed the plugin system and how it can be implemented in the SpectralVectors/RenderReminder add-on.

If you are satisfied with the POC and wish to proceed with implementing this change, once this issue is assigned to me, I'll be happy to try applying the necessary modifications.

Thank you again for your interest and support for this improvement proposal. I remain available for further discussions and collaborations.
